dist/tools/compile_test/compile_test.py

```diff
@@ -100,22 +106,24 @@ def build_all():
                                    env=subprocess_env)
 
                 results, results_with_output = tee(get_results_and_output_from(subprocess.stdout))
-                results = groupby(sorted(results), lambda (outcome, board, output): outcome)
-                results_with_output = filter(lambda (outcome, board, output): output.getvalue(), results_with_output)
-                failed_with_output = filter(lambda (outcome, board, output): 'failed' in outcome, results_with_output)
-                success_with_output = filter(lambda (outcome, board, output): 'success' in outcome, results_with_output)
+                results = groupby(sorted(results), lambda res: res[0])
+                results_with_output = list(filter(lambda res: res[2].getvalue(), results_with_output))
+                failed_with_output = list(filter(lambda res: 'failed' in res[0], results_with_output))
+                success_with_output = list(filter(lambda res: 'success' in res[0], results_with_output))
```

Comment:
Found an error with Python3 when counting the number or errors and warnings. Fixed by explicitly constructing a list from the iterator produced by the filter function. (Python2 filter produces a list directly)